Release notes for v3.6.2 (#2342)
arrayfire/arrayfire: docs/pages/release_notes.md

Release Notes {#releasenotes}
==============

v3.6.2
======

Features
--------

> - Support batching for `cond` argument in select() \PR{2243}

Reviewer: ```suggestion
- Add support for batching on the `cond` argument in select() \PR{2243}
```

> - Support broadcast batching for matmul() \PR{2315}

Reviewer: ```suggestion
- Add support for broadcasting batched matmul() \PR{2315}
```

- Support multiple nearest neighbors for nearestNeighbour() \PR{2280}
- Clamp-to-edge padding as an af_border_type option \PR{2333}

Improvements
------------
- Improved performance of morphological operations \PR{2238}

> - Changed allocation of raw pointers in backends to prevent potential memory

Reviewer: ```suggestion

```

>   leaks \PR{2265}

Reviewer: ```suggestion

```

- Fixed linking errors when compiling without Freeimage/Graphics \PR{2248}

> - Added bigobj EHsc and FS flags to CUDA_NVCC_FLAGS and test executables

Reviewer: ```suggestion

```

>   \PR{2253}

Reviewer: ```suggestion

```

> - Fixed to improve the usage of ArrayFire as a subproject \PR{2290} 

Reviewer: ```suggestion
- Improved the usage of ArrayFire as a CMake subproject \PR{2290} 
```

> - Fixed compilation issues on macOS \PR{2323}

Reviewer: ```suggestion

```

- Enabled configuration of custom library path for loading dynamic backend
  libraries \PR{2302}

Bug Fixes
---------

> - Fixed LAPACK definitions \PR{2239}

Reviewer: ```suggestion
- Fixed LAPACK Definitions and linking errors. \PR{2245} \PR{2239}
```

> - Fixed LAPACK linking errors \PR{2245}

Reviewer: ```suggestion

```

- Fixed overflow in dim4::ndims() \PR{2289}

> - Removed setDevice from af::array destructor \PR{2319}

Reviewer: ```suggestion

```

- Fixed pow() precision for integral types \PR{2305}
- Fixed issues with tile() with a large repeat dimension \PR{2307}
- Fixed svd() sub-array output on OpenCL \PR{2279}
- Fixed grid-based indexing calculation in histogram() \PR{2230}

> - Fixed bug when using an af::array for indexing \PR{2311}

Reviewer: ```suggestion
- Fixed bug in indexing when used after reorder. \PR{2311}
```

> - Incorporated [CLBlast](https://github.com/CNugteren/CLBlast) fixes for Windows

Reviewer: ```suggestion
- Fixed errors when exiting on WIndows when using CLBlast.
```

  \PR{2222}

Documentation
-------------
- Improved unwrap() documentation \PR{2301}
- Improved wrap() documentation \PR{2320}
- Improved accum() documentation \PR{2298}
- Improved tile() documentation \PR{2293}
- Clarified approx1() and approx2() indexing in documentation \PR{2287}
- Updated examples of [select()](@ref data_func_select) in detailed documentation
  \PR{2277}
- Updated lookup() examples \PR{2288}
- Updated set operations' documentation \PR{2299}

Misc
----
- `af*` libraries and dependencies directory changed to `lib64` \PR{2186}
- New arrayfire ASSERT utility functions \PR{2249} \PR{2256} \PR{2257} \PR{2263}

> - Add missing AF_WITH_LOGGING definiton \PR{2275}

Reviewer: ```suggestion

```

> - Add Volta and remove Fermi from CUDA Toolkit 9.0 builds \PR{2269}

Reviewer: ```suggestion

```

- Improve error messages in JIT \PR{2309}

> - spdlog v1.0.0 added as part of ArrayFire by default \PR{2264}

Reviewer: ```suggestion

```
